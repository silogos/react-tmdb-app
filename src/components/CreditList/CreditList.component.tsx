import { UserIcon } from "@heroicons/react/16/solid";
import { DisplayNoData } from "../Displays";
import { Credit } from "@/types/Movie.types";

export function ListNoData() {
  return <DisplayNoData />;
}

export function CreditItem({ credit }: { credit: Credit }) {
  return (
    <div key={credit.id} className="flex flex-row items-center gap-2">
      {credit.profile_path ? (
        <img
          className="size-12 sm:size-14 lg:size-16 object-cover rounded-lg"
          src={`https://media.themoviedb.org/t/p/w276_and_h350_face/${credit.profile_path}`}
        ></img>
      ) : (
        <div className="bg-slate-400 rounded-lg">
          <UserIcon
            data-testid="user-profile"
            className="size-12 sm:size-14 lg:size-16"
          />
        </div>
      )}
      <div className="flex flex-col gap-1">
        <div className="text-xs sm:text-sm lg:text-base font-bold">
          {credit.original_name}
        </div>
        <div className="text-xs sm:text-sm lg:text-base font-thin">
          {credit.character || credit.known_for_department}
        </div>
      </div>
    </div>
  );
}
