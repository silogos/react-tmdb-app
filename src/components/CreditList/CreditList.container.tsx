import { CreditListProps } from "./CreditList.type";
import { CreditItem, ListNoData } from "./CreditList.component";

function CreditList({ title, credits }: CreditListProps) {
  return (
    <section className="overflow-hidden mb-4">
      <div className="text-white mx-auto max-w-screen-xl px-8 py-2 sm:py-4 lg:py-8">
        <h2 className="text-lg sm:text-xl lg:text-2xl font-bold mb-4">
          {title} <span className="text-gray-300">({credits.length})</span>
        </h2>
        <div className="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4 gap-y-8">
          {credits.map((credit) => (
            <CreditItem key={credit.credit_id} credit={credit} />
          ))}
        </div>
        {credits.length < 1 && <ListNoData />}
      </div>
    </section>
  );
}

export default CreditList;
