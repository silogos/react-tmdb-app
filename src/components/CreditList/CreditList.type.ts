import { Credit } from "@/types/Movie.types";

export type CreditListProps = {
  title: string;
  credits: Credit[];
};
