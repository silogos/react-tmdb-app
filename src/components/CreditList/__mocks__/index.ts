export { default } from "./CreditList.container";
