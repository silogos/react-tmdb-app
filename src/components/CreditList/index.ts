export { default } from "./CreditList.component";
