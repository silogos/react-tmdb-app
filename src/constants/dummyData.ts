import { Credit, Movie, MovieDetail } from "@/types/Movie.types";

export const dummyMovies: Movie[] = [
  {
    adult: false,
    backdrop_path: "/b85bJfrTOSJ7M5Ox0yp4lxIxdG1.jpg",
    genre_ids: [28, 878, 35, 10751],
    id: 939243,
    original_language: "en",
    original_title: "Sonic the Hedgehog 3",
    overview:
      "Sonic, Knuckles, and Tails reunite against a powerful new adversary, Shadow, a mysterious villain with powers unlike anything they have faced before. With their abilities outmatched in every way, Team Sonic must seek out an unlikely alliance in hopes of stopping Shadow and protecting the planet.",
    popularity: 7535.831,
    poster_path: "/d8Ryb8AunYAuycVKDp5HpdWPKgC.jpg",
    release_date: "2024-12-19",
    title: "Sonic the Hedgehog 3",
    video: false,
    vote_average: 7.8,
    vote_count: 1078,
  },
  {
    adult: false,
    backdrop_path: "/xZm5YUNY3PlYD1Q4k7X8zd2V4AK.jpg",
    genre_ids: [28, 35],
    id: 993710,
    original_language: "en",
    original_title: "Back in Action",
    overview:
      "Fifteen years after vanishing from the CIA to start a family, elite spies Matt and Emily jump back into the world of espionage when their cover is blown.",
    popularity: 3559.004,
    poster_path: "/3L3l6LsiLGHkTG4RFB2aBA6BttB.jpg",
    release_date: "2025-01-15",
    title: "Back in Action",
    video: false,
    vote_average: 6.724,
    vote_count: 533,
  },
];

export const dummyMovie: Movie = dummyMovies[0];

export const dummyMovieDetail: MovieDetail = {
  adult: false,
  backdrop_path: "/b85bJfrTOSJ7M5Ox0yp4lxIxdG1.jpg",
  belongs_to_collection: {
    id: 720879,
    name: "Sonic the Hedgehog Collection",
    poster_path: "/fwFWhYXj8wY6gFACtecJbg229FI.jpg",
    backdrop_path: "/l5CIAdxVhhaUD3DaS4lP4AR2so9.jpg",
  },
  budget: 122000000,
  genres: [
    {
      id: 28,
      name: "Action",
    },
    {
      id: 878,
      name: "Science Fiction",
    },
    {
      id: 35,
      name: "Comedy",
    },
    {
      id: 10751,
      name: "Family",
    },
  ],
  homepage: "https://www.sonicthehedgehogmovie.com",
  id: 939243,
  imdb_id: "tt18259086",
  origin_country: ["US"],
  original_language: "en",
  original_title: "Sonic the Hedgehog 3",
  overview:
    "Sonic, Knuckles, and Tails reunite against a powerful new adversary, Shadow, a mysterious villain with powers unlike anything they have faced before. With their abilities outmatched in every way, Team Sonic must seek out an unlikely alliance in hopes of stopping Shadow and protecting the planet.",
  popularity: 3560.42,
  poster_path: "/d8Ryb8AunYAuycVKDp5HpdWPKgC.jpg",
  production_companies: [
    {
      id: 4,
      logo_path: "/gz66EfNoYPqHTYI4q9UEN4CbHRc.png",
      name: "Paramount Pictures",
      origin_country: "US",
    },
    {
      id: 333,
      logo_path: "/5xUJfzPZ8jWJUDzYtIeuPO4qPIa.png",
      name: "Original Film",
      origin_country: "US",
    },
    {
      id: 77884,
      logo_path: "/dP2lxVNctD5Cried0IWVqgrO2o9.png",
      name: "Marza Animation Planet",
      origin_country: "JP",
    },
    {
      id: 113750,
      logo_path: "/A3QVZ9Ah0yI2d2GiXUFpdlbTgyr.png",
      name: "SEGA",
      origin_country: "JP",
    },
    {
      id: 10644,
      logo_path: "/ocLZIdYJBppuCt1rhYEb2jbpt5F.png",
      name: "Blur Studio",
      origin_country: "US",
    },
    {
      id: 168701,
      logo_path: "/vWdZFT4V64CCv12D10m44duQjyg.png",
      name: "SEGA of America",
      origin_country: "US",
    },
  ],
  production_countries: [
    {
      iso_3166_1: "JP",
      name: "Japan",
    },
    {
      iso_3166_1: "US",
      name: "United States of America",
    },
  ],
  release_date: "2024-12-19",
  revenue: 422398000,
  runtime: 110,
  spoken_languages: [
    {
      english_name: "English",
      iso_639_1: "en",
      name: "English",
    },
    {
      english_name: "French",
      iso_639_1: "fr",
      name: "Français",
    },
    {
      english_name: "German",
      iso_639_1: "de",
      name: "Deutsch",
    },
    {
      english_name: "Japanese",
      iso_639_1: "ja",
      name: "日本語",
    },
    {
      english_name: "Letzeburgesch",
      iso_639_1: "lb",
      name: "",
    },
    {
      english_name: "Malagasy",
      iso_639_1: "mg",
      name: "",
    },
    {
      english_name: "Spanish",
      iso_639_1: "es",
      name: "Español",
    },
  ],
  status: "Released",
  tagline: "Try to keep up.",
  title: "Sonic the Hedgehog 3",
  video: false,
  vote_average: 7.597,
  vote_count: 725,
};

export const dummyCredits: Credit[] = [
  {
    adult: false,
    gender: 2,
    id: 27428,
    known_for_department: "Acting",
    name: "Aaron Taylor-Johnson",
    original_name: "Aaron Taylor-Johnson",
    popularity: 49.234,
    profile_path: "/pFtHhih2XEaFaD3qOFyQW6q83br.jpg",
    cast_id: 8,
    character: "Sergei Kravinoff / Kraven the Hunter",
    credit_id: "60aec3f9d29bdd002c022ce0",
    order: 0,
  },
  {
    adult: false,
    gender: 1,
    id: 1437491,
    known_for_department: "Acting",
    name: "Ariana DeBose",
    original_name: "Ariana DeBose",
    popularity: 17.829,
    profile_path: "/hEamSutadgLbVNHch3KhJ5LfPdp.jpg",
    cast_id: 14,
    character: "Calypso Ezili",
    credit_id: "621e8519f12cf4001b7f2ccf",
    order: 1,
  },
];
export const dummyCredit: Credit = dummyCredits[0];
