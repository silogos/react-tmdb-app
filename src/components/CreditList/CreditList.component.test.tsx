import { describe, it, expect } from "vitest";
import { ListNoData, CreditItem } from "./CreditList.component";
import { render } from "@testing-library/react";
import { Credit } from "@/types/Movie.types";
import { dummyCredit } from "@/constants/dummyData";

describe("CreditList Component", () => {
  it("renders ListNoData correctly", () => {
    const component = render(<ListNoData />);

    expect(component.container).toMatchSnapshot();
  });

  describe("CreditItem", () => {
    it("renders correctly", () => {
      const component = render(<CreditItem credit={dummyCredit} />);

      expect(component.getByText("Aaron Taylor-Johnson")).toBeInTheDocument();
      expect(
        component.getByText("Sergei Kravinoff / Kraven the Hunter")
      ).toBeInTheDocument();
      expect(component.container).toMatchSnapshot();
    });

    it("displays subtitle use known_for_department when character in null", () => {
      const _dummyCredit: Credit = {
        ...dummyCredit,
        character: undefined,
      };
      const component = render(<CreditItem credit={_dummyCredit} />);

      expect(component.getByText("Acting")).toBeInTheDocument();
      expect(component.container).toMatchSnapshot();
    });

    it("displays the UserIcon when profile_path in null", () => {
      const _dummyCredit: Credit = {
        ...dummyCredit,
        profile_path: undefined,
      };
      const component = render(<CreditItem credit={_dummyCredit} />);

      expect(component.getByTestId("user-profile")).toBeInTheDocument();
      expect(component.container).toMatchSnapshot();
    });
  });
});
