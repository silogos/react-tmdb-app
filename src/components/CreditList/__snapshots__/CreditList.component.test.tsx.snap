// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CreditList Component > CreditItem > displays subtitle use known_for_department when character in null 1`] = `
<div>
  <div
    class="flex flex-row items-center gap-2"
  >
    <img
      class="size-12 sm:size-14 lg:size-16 object-cover rounded-lg"
      src="https://media.themoviedb.org/t/p/w276_and_h350_face//pFtHhih2XEaFaD3qOFyQW6q83br.jpg"
    />
    <div
      class="flex flex-col gap-1"
    >
      <div
        class="text-xs sm:text-sm lg:text-base font-bold"
      >
        Aaron Taylor-Johnson
      </div>
      <div
        class="text-xs sm:text-sm lg:text-base font-thin"
      >
        Acting
      </div>
    </div>
  </div>
</div>
`;

exports[`CreditList Component > CreditItem > displays the UserIcon when profile_path in null 1`] = `
<div>
  <div
    class="flex flex-row items-center gap-2"
  >
    <div
      class="bg-slate-400 rounded-lg"
    >
      <svg
        aria-hidden="true"
        class="size-12 sm:size-14 lg:size-16"
        data-slot="icon"
        data-testid="user-profile"
        fill="currentColor"
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 8a3 3 0 1 0 0-6 3 3 0 0 0 0 6ZM12.735 14c.618 0 1.093-.561.872-1.139a6.002 6.002 0 0 0-11.215 0c-.22.578.254 1.139.872 1.139h9.47Z"
        />
      </svg>
    </div>
    <div
      class="flex flex-col gap-1"
    >
      <div
        class="text-xs sm:text-sm lg:text-base font-bold"
      >
        Aaron Taylor-Johnson
      </div>
      <div
        class="text-xs sm:text-sm lg:text-base font-thin"
      >
        Sergei Kravinoff / Kraven the Hunter
      </div>
    </div>
  </div>
</div>
`;

exports[`CreditList Component > CreditItem > renders correctly 1`] = `
<div>
  <div
    class="flex flex-row items-center gap-2"
  >
    <img
      class="size-12 sm:size-14 lg:size-16 object-cover rounded-lg"
      src="https://media.themoviedb.org/t/p/w276_and_h350_face//pFtHhih2XEaFaD3qOFyQW6q83br.jpg"
    />
    <div
      class="flex flex-col gap-1"
    >
      <div
        class="text-xs sm:text-sm lg:text-base font-bold"
      >
        Aaron Taylor-Johnson
      </div>
      <div
        class="text-xs sm:text-sm lg:text-base font-thin"
      >
        Sergei Kravinoff / Kraven the Hunter
      </div>
    </div>
  </div>
</div>
`;

exports[`CreditList Component > renders ListNoData correctly 1`] = `
<div>
  <div
    class="flex flex-row justify-center items-center rounded-lg bg-black bg-opacity-50 p-8 text-gray-300"
  >
    <svg
      aria-hidden="true"
      class="size-8 mr-2"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m11.25 11.25.041-.02a.75.75 0 0 1 1.063.852l-.708 2.836a.75.75 0 0 0 1.063.853l.041-.021M21 12a9 9 0 1 1-18 0 9 9 0 0 1 18 0Zm-9-3.75h.008v.008H12V8.25Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
    Data Not Found
  </div>
</div>
`;
